sdk/monitor/Azure.Monitor.Ingestion/tests/MonitorIngestionTestEnvironment.cs

```diff
@@ -7,18 +7,18 @@
 
 namespace Azure.Monitor.Ingestion.Tests
 {
-    public class IngestionClientTestEnvironment : TestEnvironment
+    public class MonitorIngestionTestEnvironment : TestEnvironment
     {
-        public string DCRImmutableId => GetRecordedVariable("AZURE_MONITOR_INGESTION_LOGS_DCR_RULE_ID");
+        public string DCRImmutableId => GetRecordedVariable("AZURE_MONITOR_INGESTION_LOGS_DCR_RULE_ID", options => options.IsSecret());
 
-        public string DCREndpoint => GetRecordedVariable("AZURE_MONITOR_INGESTION_DATA_COLLECTION_ENDPOINT");
+        public string DCREndpoint => GetRecordedVariable("AZURE_MONITOR_INGESTION_DATA_COLLECTION_ENDPOINT", options => options.IsSecret());
 
-        public string Ingestion_WorkspaceId => GetRecordedVariable("AZURE_MONITOR_INGESTION_INGESTION_WORKSPACE_ID");
+        public string Ingestion_WorkspaceId => GetRecordedVariable("AZURE_MONITOR_INGESTION_INGESTION_WORKSPACE_ID", options => options.IsSecret());
 
-        public string StreamName => GetRecordedVariable("AZURE_MONITOR_INGESTION_LOGS_DCR_STREAM_NAME");
+        public string StreamName => GetRecordedVariable("AZURE_MONITOR_INGESTION_LOGS_DCR_STREAM_NAME", options => options.IsSecret());
 
-        public string TableName => GetRecordedVariable("AZURE_MONITOR_INGESTION_LOGS_TABLE_NAME");
+        public string TableName => GetRecordedVariable("AZURE_MONITOR_INGESTION_LOGS_TABLE_NAME", options => options.IsSecret());
 
-        public ClientSecretCredential ClientSecretCredential => new ClientSecretCredential(GetRecordedVariable("AZURE_TENANT_ID"), GetRecordedVariable("AZURE_CLIENT_ID"), GetRecordedVariable("AZURE_CLIENT_SECRET"));
+        public ClientSecretCredential ClientSecretCredential => new ClientSecretCredential(GetRecordedVariable("AZURE_TENANT_ID", options => options.IsSecret()), GetRecordedVariable("AZURE_CLIENT_ID", options => options.IsSecret()), GetRecordedVariable("AZURE_CLIENT_SECRET", options => options.IsSecret()));
```

Comment:
I think you can use the base TestEnvironment.Credential instead. That way you won't need to store the tenantId, clientId, in the recordings as it is not used for playback anyway.